base/server/healthcheck/pki/server/healthcheck/meta/connectivity.py

```diff
@@ -0,0 +1,157 @@
+                    logger.info("Request was made but none of the certs were retrieved")
+                    yield Result(self, constants.ERROR,
+                                 msg="PKI server is up. But, unable to retrieve any certs",
+                                 serverURI=connection.serverURI,
+                                 rest_path=cert_client.cert_url)
+
+            else:
+                yield Result(self, constants.CRITICAL,
+                             msg='CA subsystem is down')
+
+        except BaseException as e:
+            logger.error("Internal server error %s", e)
+            yield Result(self, constants.CRITICAL,
+                         msg="Internal server error. Is your PKI server and LDAP up?",
+                         exception="%s" % e)
+
+
+@registry
+class DogtagKRAConnectivityCheck(CSPlugin):
+    """
+    Test basic KRA connectivity by trying to fetch the transport cert using REST endpoint.
+    """
+
+    @duration
+    def check(self):
+        if not self.instance.is_valid():
+            logging.debug('Invalid instance: %s', self.instance.name)
+            yield Result(self, constants.CRITICAL,
+                         msg='Invalid PKI instance: %s' % self.instance.name)
+            return
+
+        self.instance.load()
+
+        kra = self.instance.get_subsystem('kra')
+
+        if not kra:
+            logger.debug("No KRA configured, skipping dogtag KRA connectivity check")
+            return
+
+        try:
+            # Make a plain HTTP GET request to /kra/admin/kra/getStatus REST api end point
+            # and check if the KRA is up
+            if kra.is_ready():
+                logger.info("KRA instance is running.")
+
+                # Make a plain HTTP GET to retrieve KRA transport cert, to test that
+                # the server is up AND is able to respond back
+                connection = PKIConnection(protocol='https',
+                                           hostname='localhost',
+                                           port='8443')
+
+                system_cert_client = SystemCertClient(connection)
+
+                # This gets the KRA cert from CS.cfg via REST API. In future, the system
+                # certs will be moved into LDAP. This means that even if LDAP is down
+                # there will be a SUCCESSFUL response if KRA is running.
+                transport_cert = system_cert_client.get_transport_cert()
+
+                if transport_cert:
+
+                    if transport_cert.serial_number:
+                        logger.info("Serial number of retrieved transport cert: %s",
+                                    transport_cert.serial_number)
+                        yield Result(self, constants.SUCCESS,
+                                     serial_number=transport_cert.serial_number,
+                                     subject_dn=transport_cert.subject_dn,
+                                     status=transport_cert.status)
```

Comment:
Same thing, the data returned on successful operation might not be used for anything.